libraries/Update/src/Update.h

```diff
@@ -19,14 +20,25 @@
 #define UPDATE_ERROR_NO_PARTITION       (10)
 #define UPDATE_ERROR_BAD_ARGUMENT       (11)
 #define UPDATE_ERROR_ABORT              (12)
+#define UPDATE_ERROR_DECRYPT            (13)
 
 #define UPDATE_SIZE_UNKNOWN 0xFFFFFFFF
 
 #define U_FLASH   0
 #define U_SPIFFS  100
 #define U_AUTH    200
 
-#define ENCRYPTED_BLOCK_SIZE 16
+#define ENCRYPTED_BLOCK_SIZE       16
+#define ENCRYPTED_TWEAK_BLOCK_SIZE 32
+#define ENCRYPTED_KEY_SIZE         32
+
+#define U_AES_DECRYPT_NONE       0
```

Comment:
I would suggest to separate the option for app and spiffs and use something like:
```cpp
#define U_AES_DECRYPT_NONE       0
#define U_AES_DECRYPT_AUTO       1
#define U_AES_DECRYPT_ON         2
```

This will make the example and option selection much easier for the users

Reply:
Sound good with me, I will simplify user options. Probably just go with those three as you listed and then user can decide image tyoe(app, spiffs, ..) with command U_FLASH, U_SPIFFS when calling Upate.begin() function. I make changes & retest. Note auto only works with app image atm, using magic byte check, I will check but dont think spiffs using a special byte at begining of image.
One question, I see only latest Arduino test are successfull, version 14  or earlier fail because #include <hwcrypto/aes.h> not found. I not got earlier version to test with to see if there a simple solution.